R/conditions.R

```diff
@@ -127,6 +127,49 @@ stop_incompatible_op <- function(op, x, y, details = NULL, ..., message = NULL,
   )
 }
 
+#' @rdname vctrs-conditions
+#' @export
+stop_incompatible_size <- function(x, y,
+                                   x_size, y_size,
+                                   x_arg = "", y_arg = "",
+                                   details = NULL,
+                                   ...,
+                                   message = NULL,
+                                   .subclass = NULL) {
+  vec_assert(x)
+  vec_assert(y)
```

Comment:
Plus `vec_asssert(x_size, numeric(), 1)` etc ?

Reply:
We currently don't have a numeric type. If we're being restrictive, we should probably opt for integer rather than double.

Reply:
Perhaps we need `vec_assert_coercible()`?

Reply:
I think in that case you'd just call `vec_cast()`? (and make a separate assertion about the size) At some point we need to fully consider all of the combinations of asserting, casting to fixed type, and casting to common type, and then how to most elegantly express them.